Merge and update `about-ql-packs` with `about-codeql-packs` (#10105)
github/codeql: docs/codeql/codeql-cli/creating-and-working-with-codeql-packs.rst

The ``codeql pack init`` command creates the directory structure and configuration files for a CodeQL pack. By default, the command creates a query pack. If you want to create a library pack, you must edit the ``qlpack.yml`` file to explicitly declare the file as a library pack by including the ``library:true`` property.

> Modifying an existing legacy CodeQL pack to create a CodeQL pack

Reviewer: Perhaps we should still refer to the old-style packs as "QL packs" to avoid confusion?

```suggestion
Modifying an existing legacy QL pack to create a CodeQL pack
```

----------------------------------------------------------------

If you already have a ``qlpack.yml`` file, you can edit it manually to convert it into a CodeQL pack.

#. Edit the ``name`` property so that it matches the format ``<scope>/<name>``, where ``<scope>`` is the name of the GitHub organization or user account that you will publish to.
#. In the ``qlpack.yml`` file, include a ``version`` property with a semver identifier, as well as an optional ``dependencies`` block.

> #. Migrate the list of dependencies in ``libraryPathDependencies`` to the ``dependencies`` block. Specify the version range for each dependency. If the range is unimportant, or you are unsure of compatibility, you can specify ``"*"``, which indicates that any version is acceptable and will default to the latest version when ``codeql pack install`` is run.

Reviewer: ```suggestion
#. Migrate the list of dependencies in ``libraryPathDependencies`` to the ``dependencies`` block. Specify the version range for each dependency. If the range is unimportant, or you are unsure of compatibility, you can specify ``"*"``, which indicates that any version is acceptable and will default to the latest version when you run ``codeql pack install``.
```

> #. Migrate the list of dependencies in ``libraryPathDependencies`` to the ``dependencies`` block. Specify the version range for each dependency. If the range is unimportant, or you are unsure of compatibility, you can specify ``"*"``, which indicates that any version is acceptable, with will default to the latest version when ``codeql pack install`` is run.

Reviewer 2: ```suggestion
#. Migrate the list of dependencies in ``libraryPathDependencies`` to the ``dependencies`` block. Specify the version range for each dependency. If the range is unimportant, or you are unsure of compatibility, you can specify ``"*"``, which indicates that any version is acceptable and will default to the latest version when ``codeql pack install`` is run.
```

For more information about the properties, see ":ref:`About CodeQL packs <about-codeql-packs>`."

Adding and installing dependencies to a CodeQL pack
---------------------------------------------------
You can add dependencies on CodeQL packs using the command ``codeql pack add``. You must specify the scope, name, and (optionally) a compatible version range.

::

  codeql pack add <scope>/<name>@x.x.x <scope>/<other-name>

> If you leave off the version range, the latest version will be added. Otherwise, the latest version that satisfies the requested range will be added.

Reviewer: ```suggestion
If you don't specify a version range, the latest version will be added. Otherwise, the latest version that satisfies the requested range will be added.
```

Author: Forgot to make this change. Will push it up shortly.
